staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -247,7 +248,7 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	terminating := apiextensions.IsCRDConditionTrue(crd, apiextensions.Terminating)
 
-	crdInfo, err := r.getOrCreateServingInfoFor(crd)
+	crdInfo, err := r.getOrCreateServingInfoFor(crd.UID, crd.Name)
```

Comment:
getOrCreateServingInfoFor now has the ability to return a NotFound error. Handle that here the same way we do above

Reply:
also ensure the crdInfo.spec contains a served version for the requested api version, and if it does not, handle it the same way we do above

Reply:
updated

I assume we don't re-check the [conditions](https://github.com/kubernetes/kubernetes/blob/70b1c436576d8fe778e7f8cf8975ce89b8ade9f0/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L238-L248) because they are in status. The race in status update always exists and it's unrelated to the informer in this PR

Reply:
correct